### PR TITLE
PathCORE analysis and web DB setup scripts

### PR DESCRIPTION
**Pull request by kathyxchen**
Hi! This pull request is focused on the Python scripts for the "full" PathCORE analysis & the paper-specific shell scripts (these should be quick to review, I hope). 

More specifically:
- Shell scripts 
   - I'd review at least `ANALYSIS.sh` and one of the other `*.sh` scripts first just to get an idea of the workflow. The `analysis_<dataset>_<model>.sh` scripts are all very similar, in case anyone is interested in just modifying the parameters of one case study or something.
- Python scripts
   - Please note that `run_network_creation.py` has already been reviewed in a non-PathCORE-specific context by @tj8901nm (there has been some rearranging because I've separated the [functionality here](https://bitbucket.org/greenelab/eadage/src/d8a98586e0f4/ensemble_construction/pathway_coverage_no_crosstalk/?at=default) into different Python packages). 
   - *What you should focus on*: 
      - `run_permutation_test.py` isn't super long, but feedback on clarity of the documentation would be great!
      - the scripts prefixed with `web_` are the longest ones - they focus on organizing the metadata generated from a PathCORE analysis so that things can be inserted into MongoDB collections. 
- `README.md` Please feel free to take a look at it and comment on whether the structure makes sense? But you'll see right away that it is incredibly unfinished.

What will be included in the next PR:
- The `data` directory, which will have additional scripts for downloading things from zenodo, generating/processing datasets as needed, etc.
   - Also, the files that can be kept in the repository (pathway definitions files, the NMF models used in the PathCORE analysis)
- `figures` directory
- `jupyter-notebooks` directory
- Finished README

Please let me know if you think adding another reviewer would be appropriate! I'm hoping I'll be able to spread out the work so that no 1 individual feels like they are reviewing _too_ much all at once.

### CODE REVIEW

**Comment by gwaybio** [comment]
I haven't looked at this PR in depth yet, but one quick github fix is to make sure all commits are being associated to your profile.

It looks like some commits are coming from @kathyxchen and others are coming from katch@seas.upenn.edu. You will need to link this account to your github handle by following [these instructions](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/)

I will review the PR contents later today

File: README.md
Comment (gwaybio) [inline README.md]: looks like a thought was cut off at the end of this sentence

File: README.md
Comment (gwaybio) [inline README.md]: May want to consider adding links to each of these files

File: analysis_Paeruginosa_NMF.sh
Comment (gwaybio) [inline analysis_Paeruginosa_NMF.sh:51]: consider indenting these command args for easier readability. [an example](https://github.com/greenelab/tad_pathways_pipeline/blob/master/example_pipeline_t2d.sh)

File: analysis_Paeruginosa_eADAGE.sh
Comment (gwaybio) [inline analysis_Paeruginosa_eADAGE.sh]: can remove

File: analysis_Paeruginosa_eADAGE.sh
Comment (gwaybio) [inline analysis_Paeruginosa_eADAGE.sh]: Is there a reason to include the web application setup in the same script as the analysis?

File: run_permutation_test.py
Comment (gwaybio) [inline run_permutation_test.py]: what are `pw0` and `pw1`?

File: run_permutation_test.py
Comment (gwaybio) [inline run_permutation_test.py]: do you even need the `_ = ` reference if its not returning anything?

File: utils.py
Comment (gwaybio) [inline utils.py:22]: `pd.read_table(path_to_file, header=0, index_col=0)` will remove the need for line 23 and 24

File: utils.py
Comment (gwaybio) [inline utils.py]: why do you set `names` and `usecols`?

File: utils.py
Comment (gwaybio) [inline utils.py]: maybe `index_col=0` will help here too?

File: utils.py
Comment (gwaybio) [inline utils.py]: this looks funny. is it over 80 char?

File: constants/__init__.py
Comment (gwaybio) [inline constants/__init__.py:1]: is there a benefit of importing functions through these dictionaries rather than using simple import statements? I am curious

**Review by gwaybio — COMMENTED** [review]
General and some specific comments. Documentation is stellar all around. Agree that the README needs modification for when analysis completed

File: utils.py
Comment (kathyxchen) [reply utils.py:22]: Ah, so I had a problem here with `index_col=0` specifically on the TCGA dataset - I kept running out of memory when I specified the `index_col`. Tried looking into this more and couldn't get a satisfactory answer, but my laptop + a small EC2 instance can't handle whatever `pandas` is doing during that operation. :( It runs fine when I `set_index` separately, so even if it's not _great_ (in terms of style/verbosity) to do it this way, I'll probably leave it as is for ease of use

File: utils.py
Comment (kathyxchen) [reply utils.py]: the KEGG definitions file that I got from Jie originally doesn't have any column names (& I wanted to link the exact script/file from the eADAGE bitbucket) so to support the file format I had to both 1) name the columns and 2) specify which columns to use.

Looking at it now, if it causes more confusion than anything else, the pathway definitions files are so small that I could remove `usecols` and not impact anything

File: constants/__init__.py
Comment (kathyxchen) [reply constants/__init__.py:1]: Ah, so this is because we wanted to generalize the Python scripts (prefixed with `run_`) so that someone who cloned/forked this repo could just add a gene signature definition / pathway name shortener to these files & pass them in as arguments to the Python scripts. 
The `--signature` flag [here](https://github.com/greenelab/PathCORE-analysis/pull/1/files#diff-6e02a9e9a7cfae7f18589d6a6b2b9b97R57) pulls from that dictionary so you wouldn't have to modify `run_network_creation.py` at all.

(^ Something along those lines will be added to the README in my next update on this PR, I think)

File: run_permutation_test.py
Comment (kathyxchen) [reply run_permutation_test.py]: pathway 0 and pathway 1 -- will add an additional comment on this file

**Comment by kathyxchen** [comment]
@gwaygenomics have made updates to address your comments! I haven't added all the links in the README - going to hold off on that until after I get this PR approved/merged.

**Review by gwaybio — APPROVED** [review]

**Pull request merged**
